v3/sockjs/eventsource.go

```diff
@@ -34,6 +36,13 @@ func (h *Handler) eventSource(rw http.ResponseWriter, req *http.Request) {
 
 type eventSourceFrameWriter struct{}
 
+var escaper *strings.Replacer = strings.NewReplacer(
+	"%", url.QueryEscape("%"),
+	"\n", url.QueryEscape("\n"),
+	"\r", url.QueryEscape("\r"),
+	"\x00", url.QueryEscape("\x00"),
+)
+
 func (*eventSourceFrameWriter) write(w io.Writer, frame string) (int, error) {
-	return fmt.Fprintf(w, "data: %s\r\n\r\n", frame)
+	return fmt.Fprintf(w, "data: %s\r\n\r\n", escaper.Replace(frame))
```

Comment:
would it make sense to add a test for this change?

Reply:
Yes, but I would appreciate some guidance or examples that could help get me started. I couldn't find any existing eventsource tests that validated messages sent from the server-to-client.

I could easily add a test just of the writer, but that felt too narrow.

Thoughts? Help?

Reply:
On the other hand why not just:

```suggestion
	return fmt.Fprintf(w, "data: %s\r\n\r\n", url.QueryEscape(frame))
```

?

Reply:
@rjeczalik I think there are some inconsistencies in QueryEscape and decodeURI on browser side:

```go
package main

import (
	"net/url"
)

func main() {
	s := url.QueryEscape(" ")
	println(s)
        // output: +
}
```

Then in browser:

```
> decodeURI("+")
< '+'
```

Follow the approach used in `sockjs-node` seems the safest thing to do IMO.